pkg/controller/apischeme/apischeme_controller.go

```diff
@@ -96,5 +119,320 @@ func (r *ReconcileAPIScheme) Reconcile(request reconcile.Request) (reconcile.Res
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, err
+	}
+
+	region, err := utils.GetClusterRegion(r.client)
+	if err != nil {
+		reqLogger.Error(err, "Couldn't get the cluster owner tags")
+		SetAPISchemeStatus(reqLogger, instance, "Couldn't reconcile", "Couldn't get the cluster's AWS region", cloudingressv1alpha1.ConditionError)
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, err
+	}
+	// We expect this secret to exist in the same namespace Account CR's are created
+	// TODO: Get the region of the cluster
+	awsClient, err := awsclient.GetAWSClient(r.client, awsclient.NewAwsClientInput{
+		SecretName: config.AWSSecretName,
+		NameSpace:  config.OperatorNamespace,
+		AwsRegion:  region,
+	})
+	if err != nil {
+		reqLogger.Error(err, "Failed to get AWS client")
+		SetAPISchemeStatus(reqLogger, instance, "Couldn't reconcile", "Couldn't create an AWS client", cloudingressv1alpha1.ConditionError)
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, err
+	}
+	subnets, err := utils.GetMasterNodeSubnets(r.client)
+	if err != nil {
+		reqLogger.Error(err, "Couldn't get the subnets used by master nodes")
+		SetAPISchemeStatus(reqLogger, instance, "Couldn't reconcile", "Couldn't get the cluster's subnets", cloudingressv1alpha1.ConditionError)
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, err
+	}
+
+	clusterBaseDomain, err := utils.GetClusterBaseDomain(r.client)
+	if err != nil {
+		reqLogger.Error(err, "Couldn't obtain the cluster's base domain")
+		SetAPISchemeStatus(reqLogger, instance, "Couldn't reconcile", "Couldn't get the cluster's base domain", cloudingressv1alpha1.ConditionError)
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, err
+	}
+	masterNodeInstances, err := utils.GetClusterMasterInstances(r.client)
+	if err != nil {
+		reqLogger.Error(err, "Couldn't detect the AWS instances for master nodes")
+		SetAPISchemeStatus(reqLogger, instance, "Couldn't reconcile", "Couldn't find the cluster's AWS instances for master nodes", cloudingressv1alpha1.ConditionError)
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, err
+	}
+
+	// In theory this could return more than one VPC for master nodes. That would be
+	// odd since the security group is associated with a single VPC.
+	vpcs, err := utils.GetMasterNodeVPCs(r.client)
+	if err != nil {
+		reqLogger.Error(err, "Couldn't get the VPC in use by master nodes")
+		SetAPISchemeStatus(reqLogger, instance, "Couldn't reconcile", "Couldn't get the VPC id for master nodes", cloudingressv1alpha1.ConditionError)
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, err
+	}
+	if len(vpcs) > 1 {
+		reqLogger.Info(fmt.Sprintf("Multiple VPCs detected for master nodes. Using %s for security group", vpcs[0]))
+	}
```

Comment:
Do we want to exit reconcile here since something is obviously wrong?

Reply:
I'm not sure. It's a good discussion point. @mwoodson what do you think?